src/libraries/System.Security.Cryptography.Cose/tests/CoseSign1MessageTests.Sign.cs

```diff
@@ -0,0 +1,418 @@
+        [InlineData(true)]
+        public void SignWithIsDetached(bool isDetached)
+        {
+            byte[] content = GetDummyContent(ContentTestCase.Small);
+            ECDsa key = ECDsaKeys[ECDsaAlgorithm.ES256];
+            HashAlgorithmName algorithmName = new HashAlgorithmName(CoseTestHelpers.SHA256);
+
+            byte[] messageEncoded = CoseSign1Message.Sign(content, key, algorithmName, isDetached);
+            VerifyContentDetached(messageEncoded, content, isDetached);
+
+            messageEncoded = CoseSign1Message.Sign(content, key, algorithmName, isDetached);
+            VerifyContentDetached(messageEncoded, content, isDetached);
+
+            static void VerifyContentDetached(byte[] messageEncoded, byte[] expecteContent, bool isDetached)
+            {
+                CoseMessage messageDecoded = CoseMessage.DecodeSign1(messageEncoded);
+                if (isDetached)
+                {
+                    Assert.Null(messageDecoded.Content);
+                }
+                else
+                {
+                    Assert.True(new ReadOnlySpan<byte>(expecteContent).SequenceEqual(messageDecoded.Content.GetValueOrDefault().Span));
+                }
+            }
+        }
+
+        [Fact]
+        public void SignWithEmptyHeaderMaps()
+        {
+            // algorithm header is required in either protected or unprotected header.
+            Assert.Throws<CryptographicException>(() => CoseSign1Message.Sign(GetDummyContent(ContentTestCase.Small), GetEmptyHeaderMap(), GetEmptyHeaderMap(), ECDsaKeys[ECDsaAlgorithm.ES256]));
+        }
+
+        [Fact]
+        public void SignWithEmptyProtectedHeaderMap()
+        {
+            CoseHeaderMap @protected = GetEmptyHeaderMap();
+            CoseHeaderMap unprotected = GetHeaderMapWithAlgorithm((int)ECDsaAlgorithm.ES256);
+            Assert.NotNull(CoseSign1Message.Sign(GetDummyContent(ContentTestCase.Small), @protected, unprotected, ECDsaKeys[ECDsaAlgorithm.ES256]));
+        }
+
+        [Fact]
+        public void SignWithEmptyUnprotectedHeaderMap()
+        {
+            CoseHeaderMap @protected = GetHeaderMapWithAlgorithm((int)ECDsaAlgorithm.ES256); 
+            CoseHeaderMap unprotected = GetEmptyHeaderMap();
+            Assert.NotNull(CoseSign1Message.Sign(GetDummyContent(ContentTestCase.Small), @protected, unprotected, ECDsaKeys[ECDsaAlgorithm.ES256]));
+        }
+
+        [Theory]
+        [InlineData(false)]
+        [InlineData(true)]
+        public void SignWithCustomHeader(bool useProtectedMap)
+        {
+            byte[] content = GetDummyContent(ContentTestCase.Small);
+            AsymmetricAlgorithm key = ECDsaKeys[ECDsaAlgorithm.ES256];
+            CoseHeaderMap @protected, unprotected, mapForCustomHeader;
+            InitializeHeaders();
+
+            mapForCustomHeader.SetValue(label: new CoseHeaderLabel(42), value: 42);
+            Assert.NotNull(CoseSign1Message.Sign(content, @protected, unprotected, key));
```

Comment:
Same as the above sign-test... this test should really confirm that we see the custom header in the payload, not just that sign produced a non-null answer.